hudi-client/src/main/java/org/apache/hudi/HoodieWriteClient.java

```diff
@@ -1182,6 +1190,27 @@ private HoodieTable getTableAndInitCtx() {
     return table;
   }
 
+  /**
+   * Adds schema to config since deletes may not have schema set in the config.
+   */
+  private void addSchemaToConfig(HoodieTableMetaClient metaClient) {
+    try {
+      HoodieActiveTimeline activeTimeline = metaClient.getActiveTimeline();
+      Option<HoodieInstant> lastInstant =
+          activeTimeline.getCommitsTimeline().filterCompletedInstants().lastInstant();
+      if (lastInstant.isPresent()) {
+        HoodieCommitMetadata commitMetadata = HoodieCommitMetadata.fromBytes(
+            activeTimeline.getInstantDetails(lastInstant.get()).get(), HoodieCommitMetadata.class);
+        if (commitMetadata.getExtraMetadata().containsKey(HoodieCommitMetadata.SCHEMA_KEY)) {
+          config.setSchema(commitMetadata.getExtraMetadata().get(HoodieCommitMetadata.SCHEMA_KEY));
```

Comment:
@bvaradar : I just exposed a setSchema() to HoodieWriteConfig. If there is any other neat way to achieve this, let me know.